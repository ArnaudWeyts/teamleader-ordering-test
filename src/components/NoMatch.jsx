import React from 'react';
import { Link } from 'react-router-dom';

const NoMatch = () => (
  <div>
    Couln&apos;t find the page you were looking for...{' '}
    <span role="img" aria-label="Bandaged emoji">
      ðŸ¤•
    </span>
    <Link to="/">
      <button type="button">Take me back to safety</button>
    </Link>
  </div>
);

export default NoMatch;
