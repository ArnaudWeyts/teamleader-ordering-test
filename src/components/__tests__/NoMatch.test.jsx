import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import NoMatch from '../NoMatch';

describe('NoMatch tests', () => {
  it('renders without crashing', () => {
    shallow(<NoMatch />);
  });

  it('renders NoMatch', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/', '/random']}>
        <NoMatch />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
