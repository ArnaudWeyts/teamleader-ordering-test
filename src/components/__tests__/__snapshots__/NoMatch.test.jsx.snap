// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch tests renders NoMatch 1`] = `
<div>
  Couln't find the page you were looking for...
   
  <span
    aria-label="Bandaged emoji"
    role="img"
  >
    ðŸ¤•
  </span>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Take me back to safety
    </button>
  </a>
</div>
`;
